mods/ra/maps/monster-tank-madness/monster-tank-madness.lua

```diff
@@ -103,13 +103,32 @@ SendAlliedUnits = function()
 	Media.PlaySpeechNotification(player, "ReinforcementsArrived")
 	Utils.Do(AlliedUnits, function(table)
 		Trigger.AfterDelay(table.delay, function()
-			Reinforcements.Reinforce(player, table.types, { StartEntryPoint.Location, StartMovePoint.Location }, 18)
+			local units = Reinforcements.Reinforce(player, table.types, { StartEntryPoint.Location, StartMovePoint.Location }, 18)
+
+			Utils.Do(units, function(unit)
+				if unit.Type == "e6" then
+					Trigger.OnKilled(unit, LandingPossible)
+				end
+			end)
 		end)
 	end)
 
 	Trigger.AfterDelay(DateTime.Seconds(1), function() InitialUnitsArrived = true end)
 end
 
+LandingPossible = function()
+	if not beachReached then
+		Utils.Do(Map.ActorsInBox(Map.TopLeft, Map.BottomRight), function(a)
+			if a.Owner == player then
+				if a.Type == "spen" or a.Type == "lst" then
+					player.MarkFailedObjective(CrossRiver)
```

Comment:
Wait a moment, doesn't that mean that the mission fails now as soon as the player owns a sub pen or transport?

Also, you're calling this from the `OnKilled` trigger of the subpen, so it should be already dead by then and you don't need to check `a.Type == "spen"`.

So, all in all that should like:
```
if a.Owner == player and a.Type == "lst" then
	return
end
player.MarkFailedObjective(CrossRiver)